perf/perf_pmu.py

```diff
@@ -0,0 +1,267 @@
+                matchFound = re.search(r"\] (.*) performance", line)
+                if matchFound:
+                    self.pmu_list.append(matchFound.group(1))
+        if len(self.pmu_list) > 0:
+            self.log.info("Found PMUs %s" % self.pmu_list)
+        else:
+            self.fail("No registered PMUs found.")
+
+    def _check_kernel_config(self, config_option):
+        # This function checks the kernel configuration with the input
+        # 'config_option' a string. If the required configuration not set then
+        # test gets cancelled.
+        ret = linux_modules.check_kernel_config(config_option)
+        if ret == linux_modules.ModuleConfig.NOT_SET:
+            self.cancel("%s not set." % config_option)
+
+    def _check_file_existence(self, filelist, listdir):
+        not_found_list = []
+        for files in filelist:
+            if files not in listdir:
+                not_found_list.append(files)
+        if not_found_list:
+            self.fail('For %s, Not found files are %s' % (self.model,
+                                                          not_found_list))
+
+    def test_config_PMU_sysfs(self):
+        # This function checks for a kernel configuration option named
+        # CONFIG_PMU_SYSFS. If set then checks for file named 'mmcra' in the
+        # sysfs directory and access the file using super user as well as a
+        # normal user with out super user privileges.
+        self._check_kernel_config('CONFIG_PMU_SYSFS')
+
+        sysfs_file = "/sys/devices/system/cpu/cpu0/"
+
+        power8 = ['mmcr0', 'mmcr1', 'mmcra']
+        power9 = ['mmcr0', 'mmcr1', 'mmcra']
+        power10 = ['mmcr0', 'mmcr1', 'mmcr3', 'mmcra']
+
+        # Check for any missing files according to the model
+        self._check_file_existence(eval(self.model), os.listdir(sysfs_file))
+
+        # allow_output_check='combined' helps in combining stdout, stderr
+        output = process.system_output("cat %smmcr*" % sysfs_file,
+                                       sudo=True, shell=True,
+                                       allow_output_check='combined',
+                                       ignore_status=True)
+        # TBD: Check for any error should be the issue
+        if b'Permission denied' in output:
+            self.fail("Unable to read mmcr* files as super user.")
+
+        self._create_temp_user()
+        output = process.system_output("su - test_pmu -c 'cat %smmcr*'"
+                                       % sysfs_file, shell=True,
+                                       allow_output_check='combined',
+                                       ignore_status=True)
+        self._remove_temp_user()
+        if b'Permission denied' not in output:
+            self.fail("Able to read mmcr* files as normal user.")
+
+    def _verify_lscpu_sysfs(self, filename, value_to_verify):
+        output = process.system_output("cat %s" % filename, sudo=True,
```

Comment:
`genio` module can be used here

Reply:
I want to see the 'permission denied error' if it occurs in-case of any non-sufficient privileges.

Reply:
Prefer using the `os.stat` and `stat` module of python to achieve that. And eventually the output can still be obtained with the genio module.